### ISSUE TITLE
some questions

### ISSUE THREAD

**Issue by sweetxiaoqizi** [wvangansbeke/Unsupervised-Classification#7]
Hello, I'm very appreciate your work, and when I look over your code and find some questions.
First, your paper saying that when computing scanloss, use one image's augmented image and its neighbor will be good, but in your code, I just see using one neighbor. In my opinion, if one image has one augmented image and one neighbor then giving one batch have batch_size m and when computing scanloss, it will be m to 2m. But I just see m to m.
Second, I don't see code how to select neighbors. So, I guess it likes KNN?

**Comment by wvangansbeke** [comment]
Hi @sweetxiaoqizi,

Not completely sure what the problem is exactly but the input of the scan-loss contains a batch of anchors (size M) and a batch of corresponding neighbors (size M). After computing the dot product between the probability vectors, you get a scalar for every element in the batch (size M), which happens at [this line](https://github.com/wvangansbeke/Unsupervised-Classification/blob/master/losses/losses.py#L111) in `losses.py`.  #4 might also clarify some issues.  

The neighbors are selected by applying nearest neighbors search after training the pretext task. You can find this procedure [here](https://github.com/wvangansbeke/Unsupervised-Classification/blob/master/simclr.py#L136).

Hope this helps.

**Comment by sweetxiaoqizi** [comment]
Thanks，very clearly. Actually，I am trying write it by using tensorflow but don't work like paper.Now I will continue modify.Thanks again
